lib/rubocop/pathfinder.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module RuboCop
+  # Common methods for searching file upwards in the directory structure.
+  module Pathfinder
+    private
+
+    def files_in_path(target_path, target_file)
+      files = dirs_to_search(target_path).map do |dir|
+        File.join(dir, target_file)
+      end
+      files.select { |file| File.exist?(file) }
+    end
+
+    def dirs_to_search(target_dir)
+      dirs = []
+      Pathname.new(File.expand_path(target_dir)).ascend do |dir_pathname|
+        break if dir_pathname.to_s == @root_level
+        dirs << dir_pathname.to_s
+      end
+      dirs << Dir.home if ENV.key? 'HOME'
```

Comment:
I was thinking about this and I think I know the answer. If the code that you are running RuboCop from a sub directory other than that under home, then home will not be in the path.

Another solution to not duplicating the home directory would be to add everything to a set instead of an array.